doc/source/exclude-patterns.md

```diff
@@ -0,0 +1,33 @@
+## Introduction
+
+The optional argument `exclude_patterns` in the `in_toto_run` API, also used by `--exclude` in the ```in-toto-run``` command line tool, applies [pathspec](http://python-path-specification.readthedocs.io) to compile [gitignore-style patterns](https://git-scm.com/docs/gitignore). Artifacts (materials and products) matched by an exclude pattern are not recorded when generating link metadata. If a pattern matches a directory, all files and subdirectories are also excluded.
+
+## Pattern Formats
+  - A literal hash serves as a comment. Escape the hash with a back-slash to match a literal hash (i.e., `\#`). 
+  - A exclamation mark negates the rest of the pattern. Artifacts matched by the rest of the pattern will be the only ones recorded, and everything else will be excluded. Escape the exclamation mark with a back-slash to match a literal exclamation mark (i.e., `\!`).
+  - The slash is a directory seperator. Separators at beginning or middle (or both) of a pattern are relative to directory level of particular .gitignore file. Separators at the end of the pattern only  match directories. (e.g., `doc/frotz/` matches `doc/frotz` directory but not `a/doc/frotz` directory; however `frotz/` matches `frotz` and `a/frotz`).
+  - A single `/` does not match any file.
```

Comment:
Maybe you can move up the information from below ("Pattern beginning with a slash") here. It feels like the fit together. :)